File: algo/JS/baekjoon/15649_NandM.js
const fs = require('fs');

let [n, m] = (fs.readFileSync('./test') + '').toString().trim().split(' ');

let ans = [];
let visit = new Array(parseInt(n)).fill(false);

const back = (d, n, m) => {
  console.log(d, n, m, '****************');
  // m개의 원소까지 찼는지 확인
  if (d === m) {
    console.log(...ans);
    return;
  }
  for (let i = 0; i < n; i++) {
    console.log('i:', i, 'visit:', visit, 'ans:', ans);
    if (!visit[i]) {
      visit[i] = true;
      ans.push(i + 1);

      back(d + 1, n, m);

      ans.pop();
      visit[i] = false;
    }
  }
};

back(0, parseInt(n), parseInt(m));

File: algo/JS/baekjoon/15649_NandM.py
def back (d,n,m):
    if d==m:
        print('********'
        ,*ans)
        return
    for i in range(n):
        if not visit[i]:
            visit[i]=True
            ans.append(i+1)

            back(d+1,n,m)

            ans.pop()
            visit[i]=False
        print('i',i,'ans',ans,'visit',visit)
            

n,m=map(int,input().split())
visit=[False]*n
ans=[]
back(0,n,m)